dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/GleifApiAccessor.kt

```diff
@@ -46,7 +47,26 @@ class GleifApiAccessor(
         //val downloadLink = URL(apiText.split("\"downloadLink\":\"")[1].split("\"")[0])
         val downloadLink = URL("https://mapping.gleif.org/api/v2/isin-lei/465e4f8f-ade3-42f3-a014-e742a78e43df/download")
         logger.info("Successfully acquired download link for mapping")
-        downloadFile(downloadLink, targetFile)
+        val tempZipFile = File.createTempFile("gleif_mapping_update", ".zip")
+        downloadFile(downloadLink, tempZipFile)
+        FileUtils.copyFile(getCsvFileFromZip(tempZipFile), targetFile)
+        tempZipFile.delete()
+    }
+
+    /**
+     * Extracts CSV file from Zip file
+     * @param zipFile the zip file
+     * @return CSV file inside Zip file
+     */
+    fun getCsvFileFromZip(zipFile: File): File {
```

Comment:
This function is not api related, still. But better here that in the csv parser, I guess. Can it be private though? I do not see the problem with having this in it's own class/object. Then we would not need to make it public for testing and responsibilities would be split more logically.